### ISSUE TITLE
Notification improvements

### ISSUE THREAD

**Issue by jleclanche** [lxde/lxqt#875]
* [ ] Do not display the app name when sending from `LXQt::Notification::notify()`
* [ ] Give notifications a maximum height, or text content
* [ ] Ship a LXQt icon in lxqt-common
 * [ ] Use it in the example notifications from lxqt-config-notificationd
* [ ] Figure out why sometimes notifications have black borders instead of spacing (reproduce with Advanced Settings -> Increase spacing).

**Comment by palinek** [comment]
> Figure out why sometimes notifications have black borders instead of spacing (reproduce with Advanced Settings -> Increase spacing). #596

What is there to firgure out? Isn't it caused by the (missing) compositor?

**Comment by jleclanche** [comment]
It shouldn't happen regardless. Notifications are square; if there's no compositor we should make sure it doesn't artifact out.

**Comment by palinek** [comment]
> Notifications are square; if there's no compositor we should make sure it doesn't artifact out.

- notifications doesn't need to be rectangular, e.g. rounded corners can be defined by stylesheet (e.g. [here](https://github.com/lxde/lxqt-common/blob/master/themes/kde-plasma/lxqt-notificationd.qss#L6))
- what do you mean by *artifact out*? There is only black background widget underneath the notifications if the compositing isn't available

**Comment by jleclanche** [comment]
@palinek The fact that black background is there can be avoided. The notifications need to be displayed as separate windows, and the graphical effects need to be removed if they aren't usable.

**Comment by intialonso** [comment]
Should we remove the rounded corners from the notifications in all the themes to avoid this?
In this logic should we also remove the themes that depend on composition (Plasma, and Dark) as if they are used in a no composite environment (and this is the majority of our users, i guess) the resul is not the expected?

**Comment by agaida** [comment]
i don't agree with jerome in this point - if one decide not to use a compositor - fine, so one has to live with the consequences, we are not in the 80s of the last century anymore.

**Comment by palinek** [comment]
> The notifications need to be displayed as separate windows, and the graphical effects need to be removed if they aren't usable (eg. when compositing is off).

We already discussed that in #596 -> [comment](https://github.com/lxde/lxqt/issues/596#issuecomment-93737171), [comment](https://github.com/lxde/lxqt/issues/596#issuecomment-93743824), [comment](https://github.com/lxde/lxqt/issues/596#issuecomment-99877866) => from my point of view, it isn't worth doing it and alf's statement `if one decide not to use a compositor - fine, so one has to live with the consequences`  describes also my opinion.

**Comment by jleclanche** [comment]
Guys, I think you're misunderstanding the issue. We have rectangular notification boxes, there's no good reason why they should have black artifacts with compositing off.

And please remember a lot of our users will not have a compositor on.

**Comment by jleclanche** [comment]
And to be clear I'm not talking about the rounded corners, i'm talking about the space between the notifications.

**Comment by palinek** [comment]
> Guys, I think you're misunderstanding the issue.

Did you read the comments with the consequencies of having the notfications as standalone windows?

**Comment by jleclanche** [comment]
I missed that, sorry. I don't know, it seems very weird that we don't have them as separate windows in the first place.. anyway it's not very high priority.
